src/main/java/com/greenfox/tribesoflagopus/backend/service/RegistrationService.java

```diff
@@ -47,29 +61,80 @@
       return ResponseEntity.badRequest().body(missingAllFields);
     }
 
-    if ("occupiedUserName".equals(registerInput.getUsername())) {
+    username = registerInput.getUsername();
+    password = registerInput.getPassword();
+    setKingdomName(registerInput);
+
+    if (occupiedUserName()) {
       StatusResponse occupiedUserNameStatus = StatusResponse.builder()
           .status("error")
           .message("Username already taken, please choose an other one.")
           .build();
       return ResponseEntity.status(409).body(occupiedUserNameStatus);
     }
 
-    registerInput = setKingdomName(registerInput);
-
-    UserDto mockUser = UserDto.builder()
-        .id(1)
-        .username("Bond")
-        .kingdomId(1)
-        .build();
-    return ResponseEntity.ok().body(mockUser);
+    User user = createUserWithKingdom();
+    return createUserDto(user);
   }
 
-  private UserRegisterInput setKingdomName(UserRegisterInput registerInput) {
+  private void setKingdomName(UserRegisterInput registerInput) {
     if (registerInput.getKingdom() == null || registerInput.getKingdom().equals("")) {
-      registerInput.setKingdom(String.format("%s's kingdom", registerInput.getUsername()));
+      kingdomName = String.format("%s's kingdom", username);
+    } else {
+      kingdomName = registerInput.getKingdom();
     }
-    return registerInput;
+  }
+
+  private boolean occupiedUserName() {
+    return userRepository.existsByUsername(username);
+  }
+
+  private User createUserWithKingdom() {
+    User user = User.builder()
+        .username(username)
+        .password(password)
+        .points(0)
+        .build();
+
+    Kingdom kingdom = Kingdom.builder()
+        .name(kingdomName)
+        .build();
+
+    Location location = generateRandomLocation();
+
+    kingdom.setLocation(location);
+    location.setKingdom(kingdom);
+
+    user.setKingdom(kingdom);
+    kingdom.setUser(user);
+
+    userRepository.save(user);
+
+    return user;
+  }
+
+  private Location generateRandomLocation() {
+    Location location = new Location();
+    do {
+      location.setX(generateRandomNumber(1, 100));
+      location.setY(generateRandomNumber(1, 100));
+    } while (locationRepository.existsByXAndY(location.getX(), location.getY()));
+    return location;
+  }
+
+  private Integer generateRandomNumber(int min, int max) {
+    int random = min + (int) (Math.random() * (max + 1));
+    Integer randomNumber = Integer.valueOf(random);
+    return randomNumber;
+  }
+
+  private ResponseEntity<JsonDto> createUserDto(User user) {
+    UserDto userDto = UserDto.builder()
+        .id(user.getId())
+        .username(user.getUsername())
+        .kingdomId(user.getKingdom().getId())
+        .build();
+    return ResponseEntity.ok().body(userDto);
   }
```

Comment:
This should be refactored (move somewhere else) later, because you'll need ti create userDTO objects in other classes as well.